pyvista/core/filters/poly_data.py

```diff
@@ -1903,7 +1903,13 @@ def geodesic(
         original_ids = vtk_id_list_to_array(dijkstra.GetIdList())
 
         output = _get_output(dijkstra)
-        output["vtkOriginalPointIds"] = original_ids
+
+        try:
+            output["vtkOriginalPointIds"] = original_ids
+        except ValueError as e:
+            if str(e) != 'Number of scalars (1) must match either the number of points (0) or the number of cells (0).':
```

Comment:
Firstly, I think there might be an attribute on exceptions with the message.
Secondly, I think the real issue is `output` being an empty mesh. We should check for that I think, and raise. Maybe phrase the error as "probably not connected"  or similar.